horace_core/utilities/split_data_blocks.m

```diff
@@ -30,74 +32,122 @@
     cumulative_sum = [];
     return;
 end
-validateattributes(start_val, {'numeric'}, {'vector', 'positive'});
+validateattributes(start_pos, {'numeric'}, {'vector', 'positive'});
 validateattributes(block_sizes, {'numeric'}, {'vector', 'positive'});
-validateattributes(max_chunk_sum, {'numeric'}, {'scalar', 'positive'});
-if any(size(start_val)~=size(block_sizes))
+validateattributes(buf_size, {'numeric'}, {'scalar', 'positive'});
+if any(size(start_pos)~=size(block_sizes))
     error('HORACE:utilities:invalid_argument',...
         'The size of start_val array and size of block_sizes array have to be the same');
 end
-if size(start_val,1)>size(start_val,2) % we want it to be rows
-    start_val = start_val';    
+if size(start_pos,1)>size(start_pos,2) % we want it to be rows
+    start_pos = start_pos';
 end
 if size(block_sizes,1)>size(block_sizes,2) % we want it to be rows
-    block_sizes = block_sizes';    
+    block_sizes = block_sizes';
+end
+% glue adjacent data blocks
+cumulative_sum = cumsum(block_sizes);
+last_block_pos = block_sizes+start_pos;
+% remove border between adjacent blocks
+remove_border  = start_pos(2:end)==last_block_pos(1:end-1);
+if any(remove_border)
+    % keep boder positions, which should not be removed
```

Comment:
sorry to be picky, but please remove the comma. With the comma it means that no border positions should be removed. Without, it means that you only keep the border posittions which should not be removed.

Reply:
Russian grammar requests to highlight a subordinate clause with comma. I am instinctively follow it. But I indeed keep the keep positions which should not be removed